sound/soc/intel/common/soc-acpi-intel-sdw-mockup-match.h

```diff
@@ -0,0 +1,170 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+/*
+ * soc-acpi-intel-sdw-mockup-match.c - tables and support for SoundWire
+ * mockup device ACPI enumeration.
+ * Since these devices are generic, the tables are provided in a header
+ * file that can be included for different platforms.
+ *
+ * Copyright (c) 2021, Intel Corporation.
+ *
+ */
+
+#ifndef _SND_SOC_ACPI_INTEL_SDW_MOCKUP_MATCH
+#define _SND_SOC_ACPI_INTEL_SDW_MOCKUP_MATCH
+
+static const struct snd_soc_acpi_endpoint mockup_single_endpoint = {
+	.num = 0,
+	.aggregated = 0,
+	.group_position = 0,
+	.group_id = 0,
+};
```

Comment:
ok, I see why you do this, but personally I'd rather not allocate data in a header. Not sure what Linux style says about that or our project preferences but maybe I'd even make it a .c file and `#include` it just the same as a header. Or maybe make that a macro like
```
#define MOCKUP_ENDPOINTS \
static const struct snd_soc_acpi_endpoint mockup_single_endpoint = { \
	.num = 0, \
...
```
and then just use that macro in your .c. This would look rather ugly, but OTOH it would give you a possibility to parametrise these structures. In fact you could even define macros to define individual endpoint or device structures. I.e. this doesn't look very attractive to me but I'm not sure whether one of the proposed alternatives would be better...

Reply:
I moved all the tables to a C file and added the object to the module, exporting only 3 tables definitions in the header. much cleaner indeed. thanks for the suggestion @lyakh